lib/pure/hashes.nim

```diff
@@ -123,6 +123,33 @@ proc hiXorLo(a, b: uint64): uint64 {.inline.} =
     else:
       result = hiXorLoFallback64(a, b)
 
+when defined(js):
+  import std/jsbigints
+  import std/private/jsutils
+
+  proc noBigInt(): bool =
+    asm """`result` = typeof BigInt == 'undefined'"""
+
+  proc hiXorLoJs(a, b: JsBigInt): JsBigInt =
+    let prod = a * b
+    let mask = (big"1" shl big"64") - big"1"
```

Comment:
```suggestion
    let mask = big"0xffffffffffffffff"
```

and add a runnableExample in jsbigints:
```nim
    doAssert big"0xffffffffffffffff" == (big"1" shl big"64") - big"1"
```

(we don't have any examples with `big"0x..."` yet; this is a good one to have)

Reply:
It takes `float` too, but I did not add it,
because it meant to have a *common shared API* with the C target BigInt 
(that someone needs to build soon).

Reply:
> It takes float too, but I did not add it,

@juancarlospaco 
how so?
```
> BigInt(1.2)
Uncaught:
RangeError: The number 1.2 cannot be converted to a BigInt because it is not an integer
    at BigInt (<anonymous>)
> BigInt("1.2")
Uncaught SyntaxError: Cannot convert 1.2 to a BigInt
```

Reply:
Just like that on Node and Browser, works on my machine. :shrug:

Reply:
?

Reply:
https://tio.run/##y0osSyxOLsosKNHNy09J/f8/OT@vOD8nVS8nP13DKTPdM69Ew1LPQFPz/38A

Reply:
@juancarlospaco you're reading this wrong; js doesn't distinguish 9 vs 9.0 (unlike nim before jsgen); try `console.log(BigInt(9.1))` and it'll fail as it should